ruuvitag_sensor/decoder.py

```diff
@@ -294,6 +297,133 @@ def decode_data(self, data: str) -> Optional[SensorData5]:
+            return None
+        return round((int(data[3]) + 50000) / 100, 2)
+
+
+    def _get_pm1_ug_m3(self, data: list[str]) -> Optional[float]:
+        """Return PM 1.0, ug/m^3"""
+        if data[4] == 0xFFFF:
+            return None
+        return round(int(data[4]) * 0.1, 1)
+
+    def _get_pm25_ug_m3(self, data: list[str]) -> Optional[float]:
+        """Return PM 2.5, ug/m^3"""
+        if data[5] == 0xFFFF:
+            return None
+        return round(int(data[5]) * 0.1, 1)
+
+    def _get_pm4_ug_m3(self, data: list[str]) -> Optional[float]:
+        """Return PM 4.0, ug/m^3"""
+        if data[6] == 0xFFFF:
+            return None
+        return round(int(data[6]) * 0.1, 1)
+
+    def _get_pm10_ug_m3(self, data: list[str]) -> Optional[float]:
+        """Return PM 10.0, ug/m^3"""
+        if data[7] == 0xFFFF:
+            return None
+        return round(int(data[7]) * 0.1, 1)
+
+    def _get_co2_ppm(self, data: list[str]) -> Optional[int]:
+        """Return CO2 ppm"""
+        if data[8] == 0xFFFF:
+            return None
+        return int(data[8])
+
+    def _get_voc_index(self, data: list[str]) -> Optional[int]:
+        """Return VOC index"""
+        # VOC is 9 bits: 8 bits in data[9], LSB in bit 6 of data[14]
+        val = int(data[9]) << 1
+        if data[14] & 64:  # bit 6 of data[14]
+            val |= 1
+        if val == 0x1FF:
+            return None
+        return int(val)
+
+    def _get_nox_index(self, data: list[str]) -> Optional[int]:
+        """Return NOX index"""
+        # VOC is 9 bits: 8 bits in data[10], LSB in bit 6 of data[14]
+        val = int(data[10]) << 1
+        if data[14] & 64:  # bit 6 of data[14]
+            val |= 1
+        if val == 0x1FF:
+            return None
+        return int(val)
+
+    def _get_luminosity_lux(self, data: list[str]) -> Optional[float]:
+        """Return Luminosity Lux"""
+        lumi_bytes = bytes(data[11])
+        if lumi_bytes == b"\xff\xff\xff":
+            return None
+        lumi_val = int.from_bytes(lumi_bytes, byteorder="big")
+        return round(lumi_val * 0.01, 2)
+
+    def _get_measurementsequencenumber(self, data: ByteData) -> Optional[int]:
+        seq_bytes = bytes(data[13])
+        if seq_bytes == b"\xff\xff\xff":
+            return None
+        return int.from_bytes(seq_bytes, byteorder="big")
+
+    def _get_calibration_in_progress(self, data: ByteData) -> bool:
+        # Bit 0 of flags indicates calibration status
+        return bool(data[14] & 1)
+
+    def _get_mac(self, data: ByteData) -> str:
+        return ":".join(f"{b:02X}" for b in bytes(data[16]))
+
+    def decode_data(self, data: str) -> Optional[SensorDataE1]:
+        """
+        Decode sensor data.
+
+        Returns:
+            dict: Sensor values
+        """
+        try:
+            byte_data: ByteData = struct.unpack(">BhHHHHHHHBB3s3s3sB5s6s", bytearray.fromhex(data[:80]))
+            return {
+                "data_format": "E1",
```

Comment:
as data format can now be also string, we need to change data_format to be int or str
https://github.com/ttu/ruuvitag-sensor/blob/62ee2f6c3d8a145e030c25462fb8547d89252cb3/ruuvitag_sensor/ruuvi_types.py#L5